src/Plot/PlotController.purs

```diff
@@ -39,27 +41,50 @@ evaluateWithX expression x = value
 runCommand :: Size -> PlotCommand -> DrawCommand Unit
 runCommand canvasSize (Empty bounds) = clearAndDrawGridLines bounds
 
-runCommand canvasSize (Plot bounds expression) = plotSimpleLine canvasSize bounds $ evaluateWithX expression
+runCommand canvasSize (Plot bounds expression) = plotSimpleLine canvasSize bounds f f''
+  where
+  f = evaluateWithX expression
+
+  f'' = evaluateWithX $ simplify $ secondDifferentiate expression
 
-plotSimpleLine :: Size -> XYBounds -> (Number -> Number) -> DrawCommand Unit
-plotSimpleLine canvasSize bounds func = for_ lines (\l -> drawPlotLine l.a l.b)
+plotSimpleLine :: Size -> XYBounds -> (Number -> Number) -> (Number -> Number) -> DrawCommand Unit
+plotSimpleLine canvasSize bounds f f'' = for_ lines (\l -> drawPlotLine l.a l.b)
   where
   rangeX = bounds.xBounds.upper - bounds.xBounds.lower
 
   rangeY = bounds.yBounds.upper - bounds.yBounds.lower
 
-  points = map (toNumber >>> toCanvasPoint) $ 0 .. (floor canvasSize.width)
+  defaultRange = map (toNumber >>> toDomainX) $ 0 .. (floor canvasSize.width)
+
+  changeInGradient = map f'' defaultRange
+
+  points = map toCanvasPoint $ concat $ zipWith toRange changeInGradient defaultRange
 
   lines = zipWith (\a b -> { a, b }) points (fromMaybe [] (tail points))
 
-  toCanvasPoint :: Number -> Position
-  toCanvasPoint canvasX = { x: canvasX, y: canvasY }
+  toRange :: Number -> Number -> Array Number
+  toRange deltaGradient value =
+    if deltaGradient < (canvasSize.width / rangeX) then
```

Comment:
```suggestion
    if abs deltaGradient < (canvasSize.width / rangeX) then
```
If I understand correctly, the second derivative could be negative at this point.

Reply:
Fixed in latest commit